### PR TITLE
src/fstests/runner: add a step to build kernel

### PR DESCRIPTION
**Pull request by alexandrasp**
Build the kernel for `kvm-xfstests`.

### CODE REVIEW

**Comment by alexandrasp** [comment]
Depends on #114 

Tested locally using tarball from node generated on staging. 

```python3 src/fstests/runner.py --settings config/kernelci.conf --yaml-config kernelci/pipeline.yaml run --db-config staging.kernelci.org --node-id 62ffe50b460c032f71ff1838```

```
Node tarball with id: 62ffe50b460c032f71ff1838        from revision: a0376948741c
Waiting...
Getting kernel source tree...
Source directory: /tmp/kci/linux-kernelci-staging-mainline-staging-mainline-20220819.0
Running job...
Checking if KVM-xfstests is available...
Usage: kvm-xfstests [<OPTIONS>] smoke|full
Usage: kvm-xfstests [<OPTIONS>] <test> ...
Usage: kvm-xfstests [<OPTIONS>] -g <group> ...
Usage: kvm-xfstests [<OPTIONS>] shell|maint
Usage: kvm-xfstests [<OPTIONS>] syz <repro>

Common options are:
	-a		- Disable auto-exclude; run all tests
	-c config	- Specify a file system configuration
	-C count	- Run the specified tests multiple times
	-I image	- Use this test appliance image
	-m mountopts	- Append mount options to fs config
	-n nr_cpus	- Specify the number of cpu's
	-numa num	- Ask KVM to create <num> NUMA nodes
	-N		- Enable networking (requires root)
	-o opts		- Extra kernel command line options
	-O opts		- Extra options for test runner
	-r ram		- Specify memory to be used in megabytes
	-x group	- Exclude group of tests from running
	-X test		- Exclude test from running
	--kernel file	- Boot the specified kernel
	--initrd initrd	- Boot with the specified initrd
	--no-log	- Don't save the log file for this run
	--no-action	- Print the command to start the VM

Common file system configurations are:
	4k 1k ext3 nojournal ext3conv metacsum dioread_nolock 
	data_journal bigalloc bigalloc_1k inline

xfstest names have the form: ext4/NNN generic/NNN shared/NNN

########## KVM-xftests found. ##########
#
# configuration written to .config
#
########## Kernel Config DONE. ##########
  SYNC    include/config/auto.conf.cmd
  DESCEND objtool
  CALL    scripts/atomic/check-atomics.sh
  CALL    scripts/checksyscalls.sh
  CHK     include/generated/compile.h
Kernel: arch/x86/boot/bzImage is ready  (#1)
########## Kernel Build DONE. ##########
...done```

```

**Comment by alexandrasp** [comment]
tested OK locally and seding results to staging. 

```python3 src/fstests/runner.py --settings config/kernelci.conf --yaml-config kernelci/pipeline.yaml run --db-config staging.kernelci.org --node-id 63041ad159968ea257d758c9```

Result:

```Waiting...
Getting kernel source tree...
Source directory: /tmp/kci/linux-kernelci-staging-mainline-staging-mainline-20220822.0
Running job...
Checking if KVM-xfstests is available...
Usage: kvm-xfstests [<OPTIONS>] smoke|full
Usage: kvm-xfstests [<OPTIONS>] <test> ...
Usage: kvm-xfstests [<OPTIONS>] -g <group> ...
Usage: kvm-xfstests [<OPTIONS>] shell|maint
Usage: kvm-xfstests [<OPTIONS>] syz <repro>

Common options are:
	-a		- Disable auto-exclude; run all tests
	-c config	- Specify a file system configuration
	-C count	- Run the specified tests multiple times
	-I image	- Use this test appliance image
	-m mountopts	- Append mount options to fs config
	-n nr_cpus	- Specify the number of cpu's
	-numa num	- Ask KVM to create <num> NUMA nodes
	-N		- Enable networking (requires root)
	-o opts		- Extra kernel command line options
	-O opts		- Extra options for test runner
	-r ram		- Specify memory to be used in megabytes
	-x group	- Exclude group of tests from running
	-X test		- Exclude test from running
	--kernel file	- Boot the specified kernel
	--initrd initrd	- Boot with the specified initrd
	--no-log	- Don't save the log file for this run
	--no-action	- Print the command to start the VM

Common file system configurations are:
	4k 1k ext3 nojournal ext3conv metacsum dioread_nolock 
	data_journal bigalloc bigalloc_1k inline

xfstest names have the form: ext4/NNN generic/NNN shared/NNN

KVM-xftests found.
#
# configuration written to .config
#
Kernel config done.
...done
```

And on [staging](https://staging.kernelci.org:9088/container/c0c8dc7e594a):
```today at 11:48:3108/23/2022 03:48:31 PM UTC [INFO] 2022-08-23 15:48:31.832260  906500526520  6304f6ba59968ea257d759d1  Complete   Pass      fstests```

File: config/runtime/fstests.jinja2 [hunk 1]
```diff
@@ -24,6 +24,21 @@ class Job(BaseJob):
             print('Kernel config error:', e)
             return 'fail'
 
+    def _kernel_build(self, kdir):
+        try:
+            os.chdir(kdir)
+            result = subprocess.run("make -j$(nproc)", check=True, shell=True)
+            if result.returncode == 0:
+                if os.path.exists('arch/x86/boot/bzImage'):
```
Comment (gctucker) [inline config/runtime/fstests.jinja2:32]: I really don't think this is necessary, it's a bit like the check for the presence of the `.config` file.  If the command passed then we can return `pass`.  If it passed but there's a problem with the generated image, some tests will fail in the next step.

Comment (alexandrasp) [inline config/runtime/fstests.jinja2:32]: @gctucker do we need to check if we have an image? I mean, in case of any problem we will get an exception..

Comment (alexandrasp) [reply config/runtime/fstests.jinja2]: alright, PR is updated and code on the client side is also updated, once we create a new tarball we will config and build the kernel.

Comment (gctucker) [reply config/runtime/fstests.jinja2]: I don't think we need to check for the presence of the kernel image.

**Review by gctucker — APPROVED** [review]

**Pull request merged**